crates/sui-json-rpc/src/read_api.rs

```diff
@@ -121,6 +123,52 @@ impl SuiRpcModule for ReadApi {
 
 #[async_trait]
 impl RpcFullNodeReadApiServer for FullNodeApi {
+    async fn get_move_function_args_tagging(
+        &self,
+        package: ObjectID,
+        module: String,
+        function: String,
+    ) -> RpcResult<Vec<String>> {
+        let object_read = self
+            .state
+            .get_object_read(&package)
+            .await
+            .map_err(|e| anyhow!("{e}"))?;
+
+        let normalized = match object_read {
+            ObjectRead::Exists(_obj_ref, object, _layout) => match object.data {
+                Data::Package(p) => p.normalize().map_err(|e| anyhow!("{e}")),
+                _ => Err(anyhow!("Object is not a package with ID {}", package)),
+            },
+            _ => Err(anyhow!("Package object does not exist with ID {}", package)),
+        }?;
+
+        let identifier = Identifier::new(function.as_str()).map_err(|e| anyhow!("{e}"))?;
+        let parameters = normalized.get(&module).and_then(|m| {
+            m.exposed_functions
+                .get(&identifier)
+                .map(|f| f.parameters.clone())
+        });
+
+        Ok(match parameters {
+            Some(parameters) => Ok(parameters
+                .iter()
+                .map(|p| match p {
+                    Type::Struct {
+                        address: _,
+                        module: _,
+                        name: _,
+                        type_arguments: _,
+                    } => "TRANSFER".into(),
+                    Type::Reference(_) => "READ".into(),
+                    Type::MutableReference(_) => "WRITE".into(),
```

Comment:
Maybe "READ+WRITE"?

Reply:
Should we just return `Reference`, `MutableReference`, and `Struct` and leave the client to interpret?

Reply:
I think that could also work. But I would probably suggest "by-value" or something similar instead of `Struct`